content/contexts/experimenterDetails.tsx

```diff
@@ -0,0 +1,67 @@
+import React from "react";
+
+interface ExperimenterData {
+  publicDescription: string;
+  proposedStartDate: Date;
+  proposedEndDate: Date;
+  proposedDuration: number;
+  startDate?: Date;
+  endDate?: Date;
+  variants: Record<string, string>;
+}
+
+interface ExperimenterState {
+  data: ExperimenterData;
+}
+
+interface ExperimenterState {
+  data: ExperimenterData;
+}
+
+const initialState: ExperimenterState = {
+  data: null,
+};
+
+export const experimenterDetailsContext = React.createContext(initialState);
+const { Provider } = experimenterDetailsContext;
+
+export const ACTION_UPDATE_DATA = "UPDATE_DATA";
+
+function reducer(state, action): ExperimenterState {
+  switch (action.type) {
+    case ACTION_UPDATE_DATA:
+      return {
+        ...state,
+        data: {
+          ...state.data,
+          ...action.data,
+        },
+      };
+
+    default:
+      return state;
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export const ExperimenterDetailsProvider: React.FC<ExperimenterState> = ({
+  children,
+  data,
+}) => {
+  /** @type {[React.ReducerState<any>, React.Dispatch<React.ReducerAction<any>>]} */
+  const [state, dispatch] = React.useReducer(reducer, initialState);
```

Comment:
I think you can avoid `any` here with

```suggestion
  const [state, dispatch] = React.useReducer<Reducer<ExperimenterState, ExperimenterAction>>(reducer, initialState);
```

Though then you'd need to define `ExperimenterAction`. I did something like this in `environment.tsx`. If this is tricky, feel free to leave it as is.